packages/cli/src/metrics/index.ts

```diff
@@ -6,66 +6,113 @@ import type express from 'express';
 import promBundle from 'express-prom-bundle';
 import promClient from 'prom-client';
 import semverParse from 'semver/functions/parse';
+import { Service } from 'typedi';
 
 export { promClient };
 
-export function configureMetrics(app: express.Application) {
-	if (!config.getEnv('endpoints.metrics.enable')) {
-		return;
+@Service()
+export class MetricsService {
+	cacheHitsTotal: promClient.Counter<'cache'> | undefined;
+
+	cacheMissesTotal: promClient.Counter<'cache'> | undefined;
+
+	cacheUpdatesTotal: promClient.Counter<'cache'> | undefined;
+
+	configureMetrics(app: express.Application) {
+		if (!config.getEnv('endpoints.metrics.enable')) {
+			return;
+		}
+
+		this.setupDefaultMetrics();
+		this.setupN8nVersionMetric();
+		this.setupCacheMetrics();
+		this.setupApiMetrics(app);
+		this.mountMetricsEndpoint(app);
 	}
 
-	setupDefaultMetrics();
-	setupN8nVersionMetric();
-	setupApiMetrics(app);
-	mountMetricsEndpoint(app);
-}
+	private setupN8nVersionMetric() {
+		const n8nVersion = semverParse(N8N_VERSION || '0.0.0');
 
-function setupN8nVersionMetric() {
-	const n8nVersion = semverParse(N8N_VERSION || '0.0.0');
+		if (n8nVersion) {
+			const versionGauge = new promClient.Gauge({
+				name: config.getEnv('endpoints.metrics.prefix') + 'version_info',
+				help: 'n8n version info.',
+				labelNames: ['version', 'major', 'minor', 'patch'],
+			});
 
-	if (n8nVersion) {
-		const versionGauge = new promClient.Gauge({
-			name: config.getEnv('endpoints.metrics.prefix') + 'version_info',
-			help: 'n8n version info.',
-			labelNames: ['version', 'major', 'minor', 'patch'],
-		});
+			versionGauge.set(
+				{
+					version: 'v' + n8nVersion.version,
+					major: n8nVersion.major,
+					minor: n8nVersion.minor,
+					patch: n8nVersion.patch,
+				},
+				1,
+			);
+		}
+	}
 
-		versionGauge.set(
-			{
-				version: 'v' + n8nVersion.version,
-				major: n8nVersion.major,
-				minor: n8nVersion.minor,
-				patch: n8nVersion.patch,
-			},
-			1,
-		);
+	private setupDefaultMetrics() {
+		if (config.getEnv('endpoints.metrics.includeDefaultMetrics')) {
+			promClient.collectDefaultMetrics();
```

Comment:
do you find these default metrics useful?
so far, I've simply ignored these as we get most of that data via kubernetes or docker-swarm anyways.

Reply:
self hosted may be interested in those as well. also, they can be disabled if not required